commit 7bdcbcdb02e6ee9307a610c685549464c0bd3316 (refs/changes/41/841/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2016-11-07T10:44:13+01:00 (2 years, 11 months ago)
    
    TOOLS-1559 wrap sapiadm and manta-login to avoid RBAC break
    Reviewed by: Trent Mick <trent.mick@joyent.com>

diff --git a/scripts/headnode.sh b/scripts/headnode.sh
index 2c27d7d4..d5fa6345 100755
--- a/scripts/headnode.sh
+++ b/scripts/headnode.sh
@@ -788,14 +788,20 @@ if setup_state_not_seen "sdczones_created"; then
         mv ${file} /opt/smartdc/bin/${tool}
     done
 
-    # Copy sapiadm into the GZ from the SAPI zone.
-    zone_uuid=$(vmadm lookup tags.smartdc_role=sapi | head -n 1)
-    if [[ -n ${zone_uuid} ]]; then
-        from_dir=/zones/${zone_uuid}/root/opt/smartdc/config-agent/cmd
-        to_dir=/opt/smartdc/bin
-        rm -f ${to_dir}/sapiadm
-        ln -s ${from_dir}/sapiadm.js ${to_dir}/sapiadm
-    fi
+    # Create wrapper for sapiadm into the GZ and execute it from
+    # the config agent.
+
+    from_dir=/opt/smartdc/agents/lib/node_modules/config-agent
+    to_dir=/opt/smartdc/bin
+    rm -f ${to_dir}/sapiadm
+
+    # BEGIN BASHSTYLED
+    cat <<EOF > ${to_dir}/sapiadm
+#!/bin/bash
+exec ${from_dir}/build/node/bin/node ${from_dir}/cmd/sapiadm.js "\$@"
+EOF
+    # END BASHSTYLED
+    chmod +x ${to_dir}/sapiadm
 
     setup_state_add "sdczones_created"
 fi
diff --git a/scripts/setup_manta_zone.sh b/scripts/setup_manta_zone.sh
index 90a5ad18..3a7d8e37 100755
--- a/scripts/setup_manta_zone.sh
+++ b/scripts/setup_manta_zone.sh
@@ -169,9 +169,14 @@ function copy_manta_tools {
         rm -f ${to_dir}/manta-oneach
 
         mkdir -p /opt/smartdc/manta-deployment/log
-        # manta-login is a bash script, so we can link it directly.
-        ln -s ${from_dir}/bin/manta-login ${to_dir}/manta-login
 
+        # While manta-login is a bash script and we could link it directly,
+        # we are using a little wrapper to avoid permission issues on the GZ.-
+        cat <<EOF > ${to_dir}/manta-login
+#!/bin/bash
+exec ${from_dir}/bin/manta-login "\$@"
+EOF
+        chmod +x ${to_dir}/manta-login
         #
         # manta-adm and manta-oneach are node programs, so we must write little
         # wrappers that call the real version using the node delivered in the
